blog/2022/11-18-tagging-as-a-treat/index.md

```diff
@@ -0,0 +1,143 @@
+
+Over 70% agree that human error is the biggest issue. Be it on the console, in the UI, in templates or in Terraform config files, tags are initially being typed by people. But then we want them to be properly understood by machines.
+
+As highlighted in [this recent blog post](../10-14-a-tale-of-two-tools/index.md) going through cloud infrastructure with CLI can be tiresome and requires a lot of contextual knowledge, as the API can differ between services. This is hardly conducive to effectively keeping tags under control. In that post we also outlined how Resoto simplifies interaction with your cloud(s).
+
+Here, we are going to go through a scenario of tag maintenance on AWS with Resoto in detail:
+
+Let us consider a (somewhat simplified) scenario where an internal policy demands that all S3 Buckets and EC2 Volumes carry a tag with the key “costcenter” and a value that corresponds to a department or a project. Most developers know about this, but the information is often conveyed by word of mouth and not by looking into the policy itself. As a result, the tag is sometimes being mis-spelled.
+
+For the sake of clarity our example only covers AWS. However, Resoto treats [all resources from all cloud providers](../09-22-cloud-resources-they-have-a-lot-in-common/index.md) the same and you can easily mix and match your search queries as needed.
+
+All examples below are executed on resotoshell after resource collection. They will use the [search command](/docs/reference/cli/search-commands/search) and the [tagging commands](/docs/concepts/resource-management/tagging).
+
+## Find out about the problem
+
+Check how many resources fall under the policy:
+
+```bash title="search for all resources that are EC2 Volumes or S3 Buckets and count them"
+> search is(aws_ec2_volume) or is(aws_s3_bucket) | count
+# highlight-start
+total matched: 20
+total unmatched: 0
+# highlight-end
+```
+
+Check how many of these resources carry the correct tag by adding a condition that a tag with the key 'costcenter' exists:
+
+```bash title="add the condition that a tag with key 'costcenter' must exist"
+> search is(aws_ec2_volume) or is(aws_s3_bucket) and tags.costcenter != null | count
+# highlight-start
+total matched: 16
+total unmatched: 0
+# highlight-end
+```
+
+If you already have your suspicions about mis-spellings check how many of these resources carry a spelling variant of the tag by varying the tag-condition:
+
+```bash title="add the condition that a tag with key **CostCenter** must exist"
+> search is(aws_ec2_volume) or is(aws_s3_bucket) and tags.CostCenter != null | count
+# highlight-start
+total matched: 4
+total unmatched: 0
+# highlight-end
+```
+
+**Alternatively**, if you don’t yet have an idea about which typos might have snuck into your data, search for all tags:
+
+```bash title="search for all tags of all resources and count how often they occur"
+> search all | jq '.tags | keys | .[] | {name:.}' | flatten | count /name
```

Comment:
Why not search for aws_ec2_volume and bucket just like above?

Reply:
because I lazily copied that example 😆 fixed it